ts/private/ts_project_worker.js

```diff
@@ -399,21 +399,24 @@ function createProgram(args, initialInputs) {
     return { host, program, checkAndApplyArgs };
 
     function setTimeout(cb) {
-        return taskQueue.push(cb) - 1;
+        // NB: tsc will never clearTimeout if the index is 0 hence timeout must be truthy. :|
+        // https://github.com/microsoft/TypeScript/blob/d0bfd8caed521bfd24fc44960d9936a891744bb7/src/compiler/watchPublic.ts#L681
+        return taskQueue.push(cb);
     }
-
+ 
     function clearTimeout(i) {
-        taskQueue[i] = null;
+        delete taskQueue[i - 1];
     }
 
     function applyChanges() {
-        debuglog(`Applying changes ${taskQueue.length}`);
-        while(taskQueue.length) {
-            const task = taskQueue.shift();
+        for (let i = 0; i < taskQueue.length; i++) {
+            const task = taskQueue[i];
+            delete taskQueue[i]
             if (task) {
                 task();
             }  
         }
+        taskQueue.splice(0, taskQueue.length);
```

Comment:
How about...
```
while (0 < taskQueue.length) {
   const task = taskQueue.shift();
   ...
}
```
Then I think we can drop this line, drop the `delete` etc

Reply:
Or maybe changing the size of the array should be avoided until the end since the indexes are IDs? Maybe just leave it as is.

Could just do `taskQueue.length = 0` though.

Reply:
tasks might schedule new tasks while being executed so the array has to stay the same until the end.

Reply:
changed with `taskQueue.length = 0`